### PR TITLE
Doxygen updates for sorc/grid_tools.fd

### PR DESCRIPTION
**Pull request by lgannoaa**
doxygen documentation for sorc/grid_tools.fd #343
The grid_tools.fd included 4 directories and 12 f90 source codes.
 Changes to be committed:
       modified:   regional_esg_grid.fd/psym2.f90
       modified:   regional_esg_grid.fd/regional_esg_grid.f90
       modified:   shave.fd/shave_nc.F90

### CODE REVIEW

File: sorc/grid_tools.fd/regional_esg_grid.fd/psym2.f90 [hunk 1]
```diff
@@ -35,18 +36,16 @@ module psym2
 
 contains
 
-!=============================================================================
+!> Get the orthogonal eigenvectors, vv, and diagonal matrix of eigenvalues, oo,
+!! of the symmetric 2*2 matrix, em.
+!! @param em symmetric 2*2 matrix
+!! @param vv orthogonal eigenvectors
+!! @param oo diagonal matrix of eigenvalues
```
Comment (edwardhartnett) [inline sorc/grid_tools.fd/regional_esg_grid.fd/psym2.f90:43]: Put an author tag on each function/subroutine. If the function/subroutine doesn't have an author, use the one from the file level. If the file doesn't have one either, ask.

File: sorc/grid_tools.fd/regional_esg_grid.fd/psym2.f90 [hunk 2]
```diff
@@ -263,33 +253,30 @@ subroutine sqrtsym2d_t(x,z,zd)!                                  [sqrtsym2d_t]
...
 real(dp),dimension(2,2):: vv,oo
 integer(spi)           :: i
-!=============================================================================
 call eigensym2(em,vv,oo)
 do i=1,2; oo(i,i)=exp(oo(i,i)); enddo
 expem=matmul(vv,matmul(oo,transpose(vv)))
 end subroutine expsym2
-!=============================================================================
+
+!>
```
Comment (edwardhartnett) [inline sorc/grid_tools.fd/regional_esg_grid.fd/psym2.f90:270]: Can you write a one-sentence description of this subroutine?

File: sorc/grid_tools.fd/regional_esg_grid.fd/psym2.f90 [hunk 3]
```diff
@@ -299,19 +286,18 @@ subroutine expsym2d(x,z,zd)!                                         [expsym2]
 z=z*exp(trxh)
 end subroutine expsym2d
 
-!=============================================================================
+!>
```
Comment (edwardhartnett) [inline sorc/grid_tools.fd/regional_esg_grid.fd/psym2.f90:289]: Can you write a one-sentence description of this subroutine?

**Review by edwardhartnett — CHANGES_REQUESTED** [review]

**Comment by edwardhartnett** [comment]
Is this PR ready to go or is there still work happening? If it's ready, please make it a proper PR instead of a draft.

File: sorc/grid_tools.fd/regional_esg_grid.fd/psym2.f90 (hunk 3 above)
Comment (lgannoaa) [reply sorc/grid_tools.fd/regional_esg_grid.fd/psym2.f90:289]: There are grid_tools.fd included 4 directories and 12 f90 source codes. Still working on it

**Comment by lgannoaa** [comment]
> Is this PR ready to go or is there still work happening? If it's ready, please make it a proper PR instead of a draft.

grid_tools.fd included 4 directories and 12 f90 source codes. Still working on it.

**Comment by lgannoaa** [comment]
Hi @GeorgeGayno-NOAA,
May I know who is the author for grid_tools.fd/shave.fd/shave_nc.F90?
Thanks

**Comment by GeorgeGayno-NOAA** [comment]
> Hi @GeorgeGayno-NOAA,
> May I know who is the author for grid_tools.fd/shave.fd/shave_nc.F90?
> Thanks

Tom Black

**Comment by edwardhartnett** [comment]
@lgannoaa can we please get these comments resolved and this PR merged before you put up any more documentation PRs?

**Comment by edwardhartnett** [comment]
@lgannoaa let me know when you are done here and I can review again. Thanks!

**Comment by lgannoaa** [comment]
> @lgannoaa let me know when you are done here and I can review again. Thanks!

@edwardhartnett as you requested, this PR has been separated into multiple PR for each f90 source file. We can delete this main PR.

**Comment by edwardhartnett** [comment]
@lgannoaa in future only submit code changes in one PR. I will close this immediately.

**Pull request closed without merging**